lib/modules/manager/pip-compile/common.ts

```diff
@@ -27,12 +59,154 @@ export function getPipToolsConstraint(config: UpdateArtifactsConfig): string {
 
   return '';
 }
+
 export const constraintLineRegex = regEx(
-  /^(#.*?\r?\n)+# {4}pip-compile(?<arguments>.*?)\r?\n/,
+  /^(#.*?\r?\n)+# {4}(?<command>\S*)(?<arguments> .*?)?\r?\n/,
 );
-export const allowedPipArguments = [
+
+export const disallowedPipOptions = [
+  '--no-header', // header is required by this manager
+];
+export const optionsWithArguments = [
+  '--output-file',
+  '--extra',
+  '--extra-index-url',
+  '--resolver',
+];
+export const allowedPipOptions = [
+  '-v',
   '--allow-unsafe',
   '--generate-hashes',
-  '--no-emit-index-url',
+  '--no-emit-index-url', // TODO: handle this!!!
   '--strip-extras',
+  ...optionsWithArguments,
 ];
+
+// as commander.js is already used, we will reuse it's argument parsing capability
+const dummyPipCompile = new Command();
+dummyPipCompile
+  .argument('[sourceFile...]') // optional, so extractHeaderCommand can throw an explicit error
+  .option('--output-file <path>')
+  .option('--extra <extra...>')
+  .option('--extra-index-url <url...>')
+  .allowUnknownOption();
+
+// TODO(not7cd): test on all correct headers, even with CUSTOM_COMPILE_COMMAND
+export function extractHeaderCommand(
+  content: string,
+  fileName: string,
+): PipCompileArgs {
+  const strict: boolean = true; // TODO(not7cd): add to function params
+  const compileCommand = constraintLineRegex.exec(content);
+  if (compileCommand?.groups === undefined) {
+    throw new Error(`Failed to extract command from header in ${fileName}`);
+  }
+  logger.debug(
+    `pip-compile: found header in ${fileName}: \n${compileCommand[0]}`,
+  );
+  const command = compileCommand.groups.command;
+  const argv: string[] = [command];
+  const isCustomCommand = command !== 'pip-compile';
+  if (isCustomCommand) {
+    logger.debug(
+      `pip-compile: custom command ${command} detected (${fileName})`,
+    );
+  }
+  if (compileCommand.groups.arguments) {
+    argv.push(...split(compileCommand.groups.arguments));
+  }
+  logger.debug(
+    `pip-compile: extracted command from header: ${JSON.stringify(argv)}`,
+  );
+  for (const arg of argv) {
+    // TODO(not7cd): check for "--option -- argument" case
+    if (!arg.startsWith('-')) {
+      continue;
+    }
+    throwForDisallowedOption(arg);
+    throwForNoEqualSignInOptionWithArgument(arg);
+    if (strict) {
+      throwForUnknownOption(arg);
+    }
+  }
+
+  // Commander.parse is expecting argv[0] to be process.execPath, pass empty string as first value
+  const parsedCommand = dummyPipCompile.parse(['', ...argv]);
```

Comment:
```suggestion
  const parsedCommand = dummyPipCompile.parse([command, ...argv]);
```
pass the original command 🤷‍♂️